File: php/src/Services/BrasilAPI.php
<?php

namespace php\Services;

use php\Contracts\CepProviderInterface;

class BrasilAPI implements CepProviderInterface
{
    public function search(string $cep): array
    {
        $response = file_get_contents("https://brasilapi.com.br/api/cep/v1/{$cep}");

        $data = json_decode($response, true);

        return [
            'cep' => $data['cep'] ?? '',
            'rua' => $data['street'] ?? '',
            'bairro' => $data['neighborhood'] ?? '',
            'cidade' => $data['city'] ?? '',
            'fonte' => 'BrasilAPI'
        ];
    }
}

File: python/Services/BrasilAPI.py
from Contracts.CepProviderInterface import CepProviderInterface
import requests


class BrasilAPI(CepProviderInterface):
    def search(self, cep: str) -> dict:
        response = requests.get(f"https://brasilapi.com.br/api/cep/v1/{cep}")

        data = response.json()

        return {
            "cep": data.get("cep"),
            "rua": data.get("street"),
            "bairro": data.get("neighborhood"),
            "cidade": data.get("city"),
            "fonte": "BrasilAPI"
        }
